lib/class-vip-request-block.php

```diff
@@ -41,32 +45,36 @@ public static function ip( string $value ) {
 		// phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.UserControlledHeaders
 		if ( isset( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
 			// phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.UserControlledHeaders, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-			$position = stripos( $_SERVER['HTTP_X_FORWARDED_FOR'], $value );
-			if ( false !== $position ) {
+			$ips = array_map( 'trim', explode( ',', $_SERVER['HTTP_X_FORWARDED_FOR'] ) );
+			if ( in_array( $value, $ips, true ) ) {
 				return self::block_and_log( $value, 'x-forwarded-for' );
 			}
 		}
+
+		return false;
 	}
 
 	/**
 	 * Block by exact match of the user agent header
 	 *
 	 * @param string $user_agent target user agent to be blocked.
-	 * @return void
+	 * @return void|bool
 	 */
 	public static function ua( string $user_agent ) {
 		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
 		if ( $user_agent === $_SERVER['HTTP_USER_AGENT'] ) {
-			self::block_and_log( $user_agent, 'user-agent' );
+			return self::block_and_log( $user_agent, 'user-agent' );
 		}
+
+		return false;
 	}
 
 	/**
 	 * Block by exact match for an arbitrary header.
 	 *
 	 * @param string $header HTTP header.
 	 * @param string $value header value.
-	 * @return void
+	 * @return void|bool
```

Comment:
Seems like these functions can't return void anymore?

Reply:
@GaryJones they can, please check https://github.com/Automattic/vip-go-mu-plugins/pull/2755/files/49a367ea0b1e502ee6c3de2394bad6ce1b25856a#diff-2aaa18fe2bb071a15fb049eb6bc4dcf92755b5f1baa6171d13e26330f629b38fR108

These functions return `true` (or sometimes `false`) only when we run tests; in the production environment, `block_and_log()` will call `exit()`.